### DISCUSSION TITLE
Unsupported Hauptwerk features

### DISCUSSION THREAD

**Discussion by Penrhos1920** [GrandOrgue/OdfEdit#20] [Ideas]
When using ODFEdit to convert a Hauptwerk odf please can ODFEdit generate a file of the lines that can't be translated to GO as the feature isn't available in GO odfs.

**Comment by eturpault** [comment]
A Hauptwerk ODF contains a lot of objects, grouped by categories and linked together to describe an organ. You can have an idea about this model in the pdf document that I built based on my observations : [https://github.com/GrandOrgue/ODFEdit/tree/main/doc](url)
It cannot be a simple list of lines to describe what has not be converted to GO ODF. You can have this information differently and more readable in OdfEdit : after having converted a HW ODF, on the right of the OdfEdit interface there is a tab named "Hauptwerk objects" which permits to navigate in the tree of the ODF objects of the loaded HW sample set (double clicking on an object permits to see both its parents and children in the tree, and thus to navigate in the tree). Once an object is selected, in the central text editor of OdfEdit you can see the attributes of the selected object. The first ones starting by _ are internal attributes of OdfEdit, if the second one named _GO_uid is empty it means that this object has not be converted in the GO ODF, if it is not empty it gives the unique identifier (uid) of the corresponding GO object.

**Comment by eturpault** [comment]
Here is a GO ODF generated by OdfEdit (next version) which permits to play this sample set with the conditional switches managed. Now the stops and buttons are working properly. To have the selected stops efficient, the button HANDREG has to be enabled.
[Chemnitz-Sauer-Demo.zip](https://github.com/GrandOrgue/ODFEdit/files/11412372/Chemnitz-Sauer-Demo.zip)
